[new product] Sourcegraph (#4914)
captn3m0/endoflife.date: products/sourcegraph.md

---
title: Sourcegraph
category: server-app
tags: java-runtime
permalink: /sourcegraph

> alternate_urls:
> - /sourcegraph-code-search

Reviewer: Not sure if this is helpful. We typically provide alternate_urls for urls that users might type and expect to work (k8s, eks etc). `/sourcegraph` is already a great URL, so don't see a need here. We could maybe add `/sg`, which seems to be used in a few places, but that is too ambiguous.

```suggestion
```

releaseDateColumn: true
eolColumn: Support

identifiers:

> - purl: pkg:github/sourcegraph/sourcegraph

Reviewer: ```suggestion
- purl: pkg:github/sourcegraph/sourcegraph
- purl: pkg:docker/sourcegraph/sg
```

auto:
  methods:
  - git: https://github.com/sourcegraph/sourcegraph.git
    regex: ^v(?P<major>\d+)\.(?P<minor>\d+)\.(?P<patch>\d+)$

>     template: '{{major}}.{{minor}}.{{patch}}'

Reviewer: The default regex should work well enough.
```suggestion
```

releases:
- releaseCycle: "5"
  releaseDate: 2023-05-22
  eol: false

>   latest: "v5.3.3"

Reviewer: We ignore all version prefixes, as noted in CONTRIBUTING.md
```suggestion
  latest: "5.3.3"
```

  latestReleaseDate: 2024-03-20

>   link: https://github.com/sourcegraph/sourcegraph/releases

Reviewer: ```suggestion
  link: https://github.com/sourcegraph/sourcegraph/releases/tag/v__LATEST__
```

- releaseCycle: "4"
  releaseDate: 2022-09-22

>   eol: true

Reviewer: Based on releaseDate for 5.0

```suggestion
  eol: 2023-05-22
```

  latest: "4.5.1"
  latestReleaseDate: 2023-02-24

>   link: https://github.com/sourcegraph/sourcegraph/releases

Reviewer: The release blog seems to be a better link:
```suggestion
  link: https://sourcegraph.com/blog/release/4.5
```

---

> [Sourcegraph](https://sourcegraph.com/) is a code search and intelligence tool for developers that helps
> them find, understand, and fix code across all of their projects and dependencies.

> Only the latest version of each major release is supported.

Reviewer: While this does seem accurate (no 4.x release since 5.0 was published), we should either have a clear citation (should also be added as `releasePolicyLink`), or clearly mention that this is an estimate.

Secondly, clearly mention we are tracking self-hosted releases.

```suggestion
Sourcegraph is available in self-hosted and cloud editions. This page only tracks releases for
the self-hosted offering.

Looking at recent releases, it seems that only the latest version of each major release is supported.
```

> not include bug-fixes, or security backports. Consequentially , this page marks the older release as unsupported.

Reviewer 2: ```suggestion
not include bug-fixes or security backports. Consequentially, this page marks all but the latest
release as unsupported.
```

Reviewer: I tried a bunch of variations, but nothing as good as this one. Thanks 💯

---
title: Sourcegraph
category: server-app
tags: java-runtime
permalink: /sourcegraph

> releasePolicyLink: https://handbook.sourcegraph.com/departments/engineering/dev/process/releases/

Reviewer 2: ```suggestion
releasePolicyLink: https://handbook.sourcegraph.com/departments/engineering/dev/process/releases/
changelogTemplate: https://github.com/sourcegraph/sourcegraph/releases/tag/v__LATEST__
```

Generated links work for both 4.x and 5.x.

releaseDateColumn: true
eolColumn: Support

identifiers:
- purl: pkg:github/sourcegraph/sourcegraph
- purl: pkg:docker/sourcegraph/sg

auto:
  methods:
  - git: https://github.com/sourcegraph/sourcegraph.git

> releases:

Reviewer 2: ```suggestion
# eol(x) = releaseDate(x+1)
releases:
```

For future maintainers.

- releaseCycle: "5"

>   releaseDate: 2023-05-22

Reviewer 2: ```suggestion
  releaseDate: 2023-03-22
```

See https://github.com/sourcegraph/sourcegraph/releases/tag/v5.0.0.

  eol: false
  latest: "5.3.3"
  latestReleaseDate: 2024-03-20

>   link: https://github.com/sourcegraph/sourcegraph/releases/tag/v__LATEST__

Reviewer 2: ```suggestion
```

We could use this as the main `changelogTemplate` as it works for both 4.x and 5.x. See my previous comment.

- releaseCycle: "4"
  releaseDate: 2022-09-22
  eol: 2023-05-22
  latest: "4.5.1"
  latestReleaseDate: 2023-02-24

>   link: https://sourcegraph.com/blog/release/4.5

Reviewer 2: ```suggestion
```

The main changelogTemplate works for both releases (https://github.com/sourcegraph/sourcegraph/releases/tag/v4.5.1). See my previous comment.

---

> [Sourcegraph](https://sourcegraph.com/) is a code search and intelligence tool for developers that helps
> them find, understand, and fix code across all of their projects and dependencies.

Sourcegraph is available in self-hosted and cloud editions. This page only tracks releases for
the self-hosted offering.

> Looking at recent releases, it seems that only the latest version of each major release is supported.

Reviewer 2: ```suggestion
Looking at recent releases, it seems that only the latest release is supported.
```

Because you set the EOL date for 4.x to the release date of 5.x.

> Sourcegraph provides [enterprise support](https://about.sourcegraph.com/contact/sales) with additional features for organizations.

Reviewer 2: ```suggestion
Sourcegraph provides [enterprise support](https://sourcegraph.com/pricing) with additional features and support for organizations.
```

Because https://sourcegraph.com/pricing gives more explanation than just the contact page.
